### ISSUE TITLE
User facing invalid DAGSTER_HOME errors are not helpful

### ISSUE THREAD

**Issue by helloworld** [dagster-io/dagster#2920]
User facing errors about invalid values for the `DAGSTER_HOME` environment variable are not helpful, and sometimes reference internal functions. These error messages should be updated to explain to the user what `DAGSTER_HOME` is and how to set it.

**When `DAGSTER_HOME` is not set:**

```
raise DagsterInvariantViolationError(
dagster.core.errors.DagsterInvariantViolationError: DAGSTER_HOME is not set, check is_dagster_home_set before invoking.
```

**When `DAGSTER_HOME` does not exist:**

```
DAGSTER_HOME=/abc dagit
```

```
raise DagsterInvariantViolationError(
dagster.core.errors.DagsterInvariantViolationError: DAGSTER_HOME "/abc" is not a folder or does not exist!
```

**When DAGSTER_HOME is not an absolute path:**

```
DAGSTER_HOME=abc dagit
```

```
raise DagsterInvariantViolationError(
dagster.core.errors.DagsterInvariantViolationError: DAGSTER_HOME must be absolute path: abc
```

The error messages should be something along the lines of:

```
"The environment variable $DAGSTER_HOME is not set. Dagster requires this environment "
"variable to be set to an existing folder on your filesystem.\n"
 "You can resolve this error by exporting the environment variable. "
"For example, you can run the following command in your shell or include it in your "
"shell configuration file: "
"\texport DAGSTER_HOME=~/dagster_home"
```

**Comment by rexledesma** [comment]
Resolved by c0d43d76d09893.

**Comment by antonyjosephraj** [comment]
Could anyone help me to set DAGSTER_HOME path?
